### PR TITLE
ci: add github actions

### PR DESCRIPTION
**Pull request by sputn1ck**
This PR adds github actions to the ci. 

Compared to travis, the `rpc-check` check was added.

Tested it locally with `act`.

Travis could be removed in a further PR.

### CODE REVIEW

**Comment by bhandras** [comment]
Can we remove travis in this PR too?

**Comment by sputn1ck** [comment]
> Can we remove travis in this PR too?

Hm simply removing `travis.yml` is not enough

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            loop-${{ runner.os }}-go-${{ env.GO_VERSION }}-
+            loop-${{ runner.os }}-go-
+
+      - name: setup go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~${{ env.GO_VERSION }}'
+
+      - name: build
+        run: make build
```
Comment (guggero) [inline .github/workflows/main.yml:94]: Add `make tags=dev` here as well?

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada: 

Someone with admin rights needs to update the required checks for merging PRs in GitHub (remove Travis and add these new CI checks).

**Review by GeorgeTsagk — APPROVED** [review]
LGTM

Why was this stalled?

**Review by GeorgeTsagk — APPROVED** [review]
LGTM :robot:

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            loop-${{ runner.os }}-go-${{ env.GO_VERSION }}-
+            loop-${{ runner.os }}-go-
+
+      - name: setup go ${{ env.GO_VERSION }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: '~${{ env.GO_VERSION }}'
+
+      - name: run unit tests
+        run: make unit
```
[the remark this answers is not in the document]
Comment (sputn1ck) [reply .github/workflows/main.yml:126]: I think they are because of the commit order

[the remark this answers is not in the document]
Comment (sputn1ck) [reply .github/workflows/main.yml:126]: so github will show failed linters in the commits now apparently

**Comment by sputn1ck** [comment]
Reorded commit order, now it isn't doing the github actions 🤦

File: .github/workflows/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sputn1ck) [reply .github/workflows/main.yml:126]: changed commit order, now it's gone

**Pull request merged**